File: interval.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    int n;
    cin >> n;
    int count = 0;
    pair<int,int> p = make_pair(0,0);
    vector<int>s(n+1);
    vector<int>e(n+1);
    vector<pair<int,int>>v(n+1);
    for (int i = 1;i<=n;i++)cin >> s[i];
    for (int i = 1;i<=n;i++)cin >> e[i];
    for (int i = 1;i<=n;i++){v[i].first = e[i];v[i].second = s[i];}
    sort(v.begin(),v.end());
    for (int i = 1;i<=n;i++){
        if (v[i].second < p.first)continue;
        count++;
        p.first = v[i].first;
        p.second = v[i].second;
    }
    cout << count;
    return 0;
}

File: interval.py
n = int(input())
s = [int(e) for e in input().split()]
end = [int(e) for e in input().split()]
ending = [(end[i],i) for i in range(n)]
ending.sort()
t = ending[0][0]
count = 1
for i in range(1,n):
    if (t > s[ending[i][1]]):continue
    t = ending[i][0]
    count+=1
print(count)